tests/pytests/test_config.py

```diff
@@ -1639,3 +1647,194 @@ def testDeprecatedConfigParamMessage():
+        if immutable:
+            continue
+        # Test true (yes)
+        checkConfigChange(env, configName, argName, 'yes', defaultConfigDict)
+
+        # Test false (no)
+        checkConfigChange(env, configName, argName, 'no', defaultConfigDict)
+
+        # Reset to default value
+        env.expect('CONFIG', 'SET', configName, defaultValue).ok()
+        currentConfigDict = getConfigDict(env)
+        env.assertEqual(currentConfigDict, defaultConfigDict)
+
+def testConfigIndependence_min_values():
+    """Test that changing one configuration value doesn't affect other configuration values"""
+    env = Env(noDefaultModuleArgs=True)
+    # set all numeric configs to min value
+    for configName, argName, _, minValue, maxValue, immutable, clusterConfig in numericConfigs:
+        if immutable:
+            continue
+        if clusterConfig:
+            if not env.isCluster():
+                continue
+        env.expect('CONFIG', 'SET', configName, minValue).ok()
+    # set all boolean configs to false (no)
+    for configName, argName, _, immutable, _ in booleanConfigs:
+        if immutable:
+            continue
+        env.expect('CONFIG', 'SET', configName, 'no').ok()
+    minValueConfigDict = getConfigDict(env)
+
+    for configName, argName, _, minValue, maxValue, immutable, clusterConfig in numericConfigs:
+        if immutable:
+            continue
+        if clusterConfig:
+            if not env.isCluster():
+                continue
+
+        # Test max value. Skip for search-conn-per-shard because it may open too many connections
+        if configName != 'search-conn-per-shard':
+            checkConfigChange(env, configName, argName, maxValue, minValueConfigDict)
+
+        # Reset to min value
+        env.expect('CONFIG', 'SET', configName, minValue).ok()
+        currentConfigDict = getConfigDict(env)
+        env.assertEqual(currentConfigDict, minValueConfigDict)
+
+    for configName, argName, _, immutable, _ in booleanConfigs:
+        if immutable:
+            continue
+
+        # Test true (yes)
+        checkConfigChange(env, configName, argName, 'yes', minValueConfigDict)
+
+        # Reset to false (no)
+        env.expect('CONFIG', 'SET', configName, 'no').ok()
+        currentConfigDict = getConfigDict(env)
+        env.assertEqual(currentConfigDict, minValueConfigDict)
+
+def testConfigIndependence_max_values():
+    """Test that changing one configuration value doesn't affect other configuration values"""
+    env = Env(noDefaultModuleArgs=True)
+    # set all numeric configs to max value
+    for configName, argName, _, minValue, maxValue, immutable, clusterConfig in numericConfigs:
+        if immutable:
+            continue
+        if clusterConfig:
+            if not env.isCluster():
+                continue
+        if configName == 'search-conn-per-shard': # Skip search-conn-per-shard because it may open too many connections
```

Comment:
But to test config independence, shouldn't be better to test it with an smaller value?